libs/parallelism/algorithms/include/hpx/parallel/algorithms/remove.hpp

```diff
@@ -354,3 +456,95 @@ namespace hpx { namespace parallel { inline namespace v1 {
             std::forward<Proj>(proj));
     }
 }}}    // namespace hpx::parallel::v1
+
+namespace hpx {
+    ///////////////////////////////////////////////////////////////////////////
+    // CPO for hpx::remove_if
+    HPX_INLINE_CONSTEXPR_VARIABLE struct remove_if_t final
+      : hpx::functional::tag<remove_if_t>
+    {
+        // clang-format off
+        template <typename FwdIter,
+            typename Pred, HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<FwdIter>::value
+            )>
```

Comment:
Could we have a check for the validity of the predicate here as well, please (same for other overloads below):

```suggestion
        template <typename FwdIter, typename Pred, 
            HPX_CONCEPT_REQUIRES_(
                hpx::traits::is_iterator<FwdIter>::value &&
                hpx::is_invocable_v<Pred,
                    typename std::iterator_traits<FwdIter>::value_type
               >
           )>
```